### PR TITLE
[Driver] Make -fintelfpga flag imply -fsycl.

### PR DESCRIPTION
**Pull request by srividya-sundaram**
Make `-fintelfpga` imply `-fsycl` since `-fintelfpga` already implies `-fsycl-targets`

### CODE REVIEW

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -1104,7 +1104,12 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
   Arg *SYCLAddTargets =
       getArgRequiringSYCLRuntime(options::OPT_fsycl_add_targets_EQ);
   Arg *SYCLLink = getArgRequiringSYCLRuntime(options::OPT_fsycl_link_EQ);
-  Arg *SYCLfpga = getArgRequiringSYCLRuntime(options::OPT_fintelfpga);
+  Arg *SYCLfpga = C.getInputArgs().getLastArg(options::OPT_fintelfpga);
+
+  // Make -fintelfpga flag imply -fsycl.
+  if (SYCLfpga && !HasValidSYCLRuntime) {
+    HasValidSYCLRuntime = true;
+  }
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:1112]:
```suggestion
  if (SYCLfpga && !HasValidSYCLRuntime)
    HasValidSYCLRuntime = true;
```
format nit

File: clang/lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -6122,6 +6128,11 @@ class OffloadingActionBuilder final {
       bool HasSYCLTargetsOption =
           SYCLAddTargets || SYCLTargets || SYCLLinkTargets;
 
+      // Make -fintelfpga flag imply -fsycl.
+      if (SYCLfpga && !HasValidSYCLRuntime) {
+        HasValidSYCLRuntime = true;
+      }
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:6134]:
```suggestion
      if (SYCLfpga && !HasValidSYCLRuntime)
        HasValidSYCLRuntime = true;
```
format nit

**Comment by srividya-sundaram** [comment]
SYCL-Unit :: Extensions/./ExtensionsTests/CommandGraphTest/GetProfilingInfoExceptionCheck failure seems un-related to this change.

File: clang/lib/Driver/Driver.cpp [hunk 3]
```diff
@@ -1093,7 +1093,12 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
   Arg *SYCLAddTargets =
       getArgRequiringSYCLRuntime(options::OPT_fsycl_add_targets_EQ);
   Arg *SYCLLink = getArgRequiringSYCLRuntime(options::OPT_fsycl_link_EQ);
-  Arg *SYCLfpga = getArgRequiringSYCLRuntime(options::OPT_fintelfpga);
+  Arg *SYCLfpga = C.getInputArgs().getLastArg(options::OPT_fintelfpga);
+
+  // Make -fintelfpga flag imply -fsycl.
+  if (SYCLfpga && !HasValidSYCLRuntime)
+    HasValidSYCLRuntime = true;
+
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:1101]: huh, I just realized this - you can just add the `OPT_fintelfpga` check to the bool setting at line 1071.

Comment (srividya-sundaram) [reply clang/lib/Driver/Driver.cpp:1101]: At L1080 - `HasValidSYCLRuntime` bool is also used to diagnose if SYCL offloading wasn't enabled for SYCL-specific options like `-fsycl-link-targets` , `-fsycl-add-targets`, `-fsycl-targets`,` -fsycl-link` and `-fsycl-host-compiler` and throw the error : 

error: 'sycl-specific-options-from-list-above' must be used in conjunction with '-fsycl' to enable offloading

Comment (mdtoguchi) [reply clang/lib/Driver/Driver.cpp:1101]: okay - that would imply moving the change up is the right one.  We only want to diagnose these situations when a SYCL enabling option isn't passed and is needed.

File: clang/lib/Driver/Driver.cpp [hunk 4]
```diff
@@ -1073,6 +1073,12 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
                                false) ||
       C.getInputArgs().hasArg(options::OPT_fsycl_device_only);
 
+  Arg *SYCLfpga = C.getInputArgs().getLastArg(options::OPT_fintelfpga);
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:1076]: Some inconsistency - you use hasArg and getLastArg (below line 6163)

**Review by mdtoguchi — APPROVED** [review]

**Comment by srividya-sundaram** [comment]
@intel/llvm-gatekeepers Hi, this is ready for merge. Thanks.

**Pull request merged**